dialects/mysql/src/main/kotlin/app/cash/sqldelight/dialects/mysql/MySqlTypeResolver.kt

```diff
@@ -63,6 +63,7 @@ internal class MySqlTypeResolver(
     "unix_timestamp" -> IntermediateType(TEXT)
     "to_seconds" -> IntermediateType(INTEGER)
     "json_arrayagg" -> IntermediateType(TEXT)
+    "date_add", "date_sub" -> encapsulatingType(exprList, TEXT, MySqlType.TIME, MySqlType.DATE, MySqlType.DATETIME, MySqlType.TIMESTAMP)
```

Comment:
Not fully sure about how this works, i.e. should it just return `IntermediateType(TEXT)`?

Reply:
you give it a list of types where if you have two types in the expression list it will pick the type closer to the front of the list. So I think this is good. I mean if tests pass you probably did it right

Reply:
Looks like the integration test is failing but I can't see what the failure is, and I can't run it locally (something about docker not being detected)

Reply:
yea you need docker to run the integration tests, it's pretty easy to set up if you're keane to try it locally.

You can see test reports on CI by downloading the build report at the bottom [here](https://github.com/cashapp/sqldelight/actions/runs/2158064716)

Reply:
I have docker installed but I didn't want to go down the rabbit hole debugging why it wasn't working 😅

In any case I think I misunderstood how the functions work. [It looks like the rules for what gets returned might be a little too complicated for SQLDelight to infer.](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-add)

The options I can think of are:

1. Return `TEXT`
2. If it's possible to rewrite the arguments, always casting it to a `DATETIME` will ensure that a `DATETIME` is always returned